cupy/cuda/stream.pyx

```diff
@@ -471,8 +474,13 @@ class Stream(_BaseStream):
             ptr = runtime.streamPerThread
             device_id = -1
         elif non_blocking:
-            ptr = runtime.streamCreateWithFlags(runtime.streamNonBlocking)
-            device_id = runtime.getDevice()
+            if priority:
+                ptr = runtime.streamCreateWithPriority(
+                    runtime.streamNonBlocking, priority)
+                device_id = runtime.getDevice()
+            else:
+                ptr = runtime.streamCreateWithFlags(runtime.streamNonBlocking)
+                device_id = runtime.getDevice()
         else:
             ptr = runtime.streamCreate()
             device_id = runtime.getDevice()
```

Comment:
This treatment is incorrect. Here, your `priority` argument gets ignored if `non_blocking` is `False`, but it should not be. I think the logic should be modified to
```python
if null:
    ...
elif ptds:
    ...
else:
    if priority is not None and non_blocking:
        ptr = runtime.streamCreateWithPriority(
            runtime.streamNonBlocking, priority)
    elif priority is not None and not non_blocking:
        ptr = runtime.streamCreateWithPriority(
            runtime.streamDefault, priority)
    elif priority is None and non_blocking:
        ptr = runtime.streamCreateWithFlags(
            runtime.streamNonBlocking)
    else:  # priority is None and not non_blocking
        ptr = runtime.streamCreate()
    device_id = runtime.getDevice()
```
(It might be possible to combine the last two branches and further simplify things. I didn't try.)

Reply:
Thank you so much for letting me know! I misunderstood the requirements of `cudaStreamCreateWithPriority`, I should've taken a better look at `cudaStreamCreateWithFlags` and the valid values `flags` can take.

I'll modify the logic right away!

Also, from what I can tell, it might not be possible to combine the last two branches to simplify things, since we have three different functions to use, and one of them takes two different parameters, totalling 4 cases. (Unless, you mean that you want me to use nested if-else statements instead of multiple elif statements, which I don't think you do, since elifs make the logic handling much clearer imo.)

Reply:
> Also, from what I can tell, it might not be possible to combine the last two branches to simplify things, since we have three different functions to use, and one of them takes two different parameters, totalling 4 cases.

I mean, we can always do this to combine the 4 cases
```python
if priority is None:
    priority = 0  # default
flag = runtime.streamNonBlocking if non_blocking else runtime.streamDefault
ptr = runtime.streamCreateWithFlags(flag, priority)    
```
Since you're to add the `Get{Flags,Priority}` APIs, could you verify if this can be done? If so we can keep the change here clean.

Reply:
:0

Will verify and make another PR!